test/controllers/agent/business/OverseasPropertyCommencementDateControllerSpec.scala

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.agent.business
+
+import java.time.LocalDate
+
+import config.featureswitch.FeatureSwitch.ReleaseFour
+import config.featureswitch.FeatureSwitching
+import controllers.agent.AgentControllerBaseSpec
+import forms.agent.OverseasPropertyCommencementDateForm
+import models.DateModel
+import models.common.{IncomeSourceModel, OverseasPropertyCommencementDateModel}
+import play.api.http.Status
+import play.api.mvc.{Action, AnyContent, Result}
+import play.api.test.Helpers._
+import services.agent.mocks.MockAgentAuthService
+import services.mocks.MockSubscriptionDetailsService
+import uk.gov.hmrc.http.cache.client.CacheMap
+import utilities.SubscriptionDataKeys.OverseasPropertyCommencementDate
+import utilities.TestModels.{testCacheMap, testIncomeSourceBoth, testIncomeSourceOverseasProperty}
+
+import scala.concurrent.Future
+
+class OverseasPropertyCommencementDateControllerSpec extends AgentControllerBaseSpec
+  with MockSubscriptionDetailsService with MockAgentAuthService with FeatureSwitching {
+
+  override val controllerName: String = "OverseasPropertyCommencementDateController"
+  override val authorisedRoutes: Map[String, Action[AnyContent]] = Map(
+    "show" -> TestOverseasPropertyCommencementDateController$.show(isEditMode = false),
+    "submit" -> TestOverseasPropertyCommencementDateController$.submit(isEditMode = false)
+  )
+
+  object TestOverseasPropertyCommencementDateController$ extends OverseasPropertyCommencementDateController(
```

Comment:
```suggestion
  object TestOverseasPropertyCommencementDateController extends OverseasPropertyCommencementDateController(
```
can you rename this and it's usages to take out the $